spring-security-oauth/src/main/java/org/springframework/security/oauth/provider/filter/CoreOAuthProviderSupport.java

```diff
@@ -1,268 +1,310 @@
-      url = request.getRequestURL().toString();
-    }
-    url = normalizeUrl(url);
-    url = oauthEncode(url);
+		String method = request.getMethod().toUpperCase();
+		return method + '&' + url + '&' + oauthEncode(queryString.toString());
+	}
 
-    String method = request.getMethod().toUpperCase();
-    return new StringBuilder(method).append('&').append(url).append('&').append(oauthEncode(queryString.toString())).toString();
-  }
+	/**
+	 * Normalize the URL for use in the signature. The OAuth spec says the URL protocol and host are to be lower-case,
+	 * and the query and fragments are to be stripped.
+	 *
+	 * @param url The URL.
+	 * @return The URL normalized for use in the signature.
+	 */
+	protected String normalizeUrl(String url) {
+		try {
+			URL requestURL = new URL(url);
+			StringBuilder normalized = new StringBuilder(requestURL.getProtocol().toLowerCase()).append("://").append(requestURL.getHost().toLowerCase());
+			if ((requestURL.getPort() >= 0) && (requestURL.getPort() != requestURL.getDefaultPort())) {
+				normalized.append(":").append(requestURL.getPort());
+			}
+			normalized.append(requestURL.getPath());
+			return normalized.toString();
+		}
+		catch (MalformedURLException e) {
+			throw new IllegalStateException("Illegal URL for calculating the OAuth signature.", e);
+		}
+	}
 
-  /**
-   * Normalize the URL for use in the signature. The OAuth spec says the URL protocol and host are to be lower-case,
-   * and the query and fragments are to be stripped.
-   *
-   * @param url The URL.
-   * @return The URL normalized for use in the signature.
-   */
-  protected String normalizeUrl(String url) {
-    try {
-      URL requestURL = new URL(url);
-      StringBuilder normalized = new StringBuilder(requestURL.getProtocol().toLowerCase()).append("://").append(requestURL.getHost().toLowerCase());
-      if ((requestURL.getPort() >= 0) && (requestURL.getPort() != requestURL.getDefaultPort())) {
-        normalized.append(":").append(requestURL.getPort());
-      }
-      normalized.append(requestURL.getPath());
-      return normalized.toString();
-    }
-    catch (MalformedURLException e) {
-      throw new IllegalStateException("Illegal URL for calculating the OAuth signature.", e);
-    }
-  }
+	/**
+	 * Loads the significant parameters (name-to-value map) that are to be used to calculate the signature base string.
+	 * The parameters will be encoded, per the spec section 9.1.
+	 *
+	 * @param request The request.
+	 * @return The significant parameters.
+	 */
+	protected SortedMap<String, SortedSet<String>> loadSignificantParametersForSignatureBaseString(HttpServletRequest request) {
+		// first collect the relevant parameters...
+		SortedMap<String, SortedSet<String>> significantParameters = new TreeMap<String, SortedSet<String>>();
+		// first pull from the request...
+		Enumeration parameterNames = request.getParameterNames();
```

Comment:
Yeah you're right that's two lookups it could improve readability but it is at expenses of performance, I'm curios about the second concern, What is the Google Compatibility Test?